Docker image builds fail because of a maven version

### Jenkins and plugins versions report

Or at least, that's what I think I can guess from this log:

```
Error:  Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.5.0:enforce (display-info) on project plugin-modernizer-parent-pom: 
Error:  Rule 1: org.apache.maven.enforcer.rules.version.RequireMavenVersion failed with message:
Error:  3.9.6 to use Maven plugins using JDK17 bytecode.
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error: Process completed with exit code 1.
```

### What Operating System are you using (both controller, and any agents involved in the problem)?

The latest Ubuntu version in the runner.
I've seen lots of people complaining about this latest image, but this may be unrelated.

### Reproduction steps

Modify anything in the Docker context.

### Expected Results

No error.

### Actual Results

```
Error:  Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.5.0:enforce (display-info) on project plugin-modernizer-parent-pom: 
Error:  Rule 1: org.apache.maven.enforcer.rules.version.RequireMavenVersion failed with message:
Error:  3.9.6 to use Maven plugins using JDK17 bytecode.
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error: Process completed with exit code 1.
```

### Anything else?

_No response_

### Are you interested in contributing a fix?

Yes

--- a/.github/workflows/github-docker-registry-push.yml
+++ b/.github/workflows/github-docker-registry-push.yml
@@ -55,6 +55,9 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-maven-
 
+      - name: Install the right Maven version
+        uses: s4u/setup-maven-action@v1.15.0
+
       # Build the project with Maven and install dependencies in the Docker context
       - name: Build with Maven and install dependencies in the Docker context
         run: mvn --version && mvn clean install && cp -vraxu ~/.m2 .m2

--- a/.github/workflows/github-docker-registry-push.yml
+++ b/.github/workflows/github-docker-registry-push.yml
@@ -57,7 +57,7 @@ jobs:
 
       # Build the project with Maven and install dependencies in the Docker context
       - name: Build with Maven and install dependencies in the Docker context
-        run: mvn clean install && cp -vraxu ~/.m2 .m2
+        run: mvn --version && mvn clean install && cp -vraxu ~/.m2 .m2
         env:
           MAVEN_CACHE: ${{ env.MAVEN_CACHE_PATH }}
 

--- a/.github/workflows/github-docker-registry-push.yml
+++ b/.github/workflows/github-docker-registry-push.yml
@@ -60,7 +60,7 @@ jobs:
 
       # Build the project with Maven and install dependencies in the Docker context
       - name: Build with Maven and install dependencies in the Docker context
-        run: mvn --version && mvn clean install && cp -vraxu ~/.m2 .m2
+        run: mvn clean install && cp -vraxu ~/.m2 .m2
         env:
           MAVEN_CACHE: ${{ env.MAVEN_CACHE_PATH }}
 

